lib/pat.c

```diff
@@ -136,6 +136,19 @@ pat_key_storage_size(uint32_t key_size)
   return pat_key_is_embeddable(key_size) ? 0 : key_size;
 }
 
+static inline grn_id
+pat_node_get_right(grn_pat *pat, pat_node_common *n)
+{
+  if (n) {
```

Comment:
Hmm. Should we check this here?

I think that passing non-`NULL` `n` is a responsibility of caller. If we do this here, caller can't know whether returned `GRN_ID_NIL` means invalid `n` or no right node.

Reply:
I see.
I removed `NULL` check in this function.